dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/EventPersister.java

```diff
@@ -172,63 +169,118 @@ private void handleDataValues(
     payloadDataValues.forEach(
         dv -> {
           DataElement dataElement = preheat.getDataElement(dv.getDataElement());
-          checkNotNull(
-              dataElement,
-              "Data element should never be NULL here if validation is enforced before commit.");
 
           // EventDataValue.dataElement contains a UID
           EventDataValue eventDataValue = dataValueDBMap.get(dataElement.getUid());
```

Comment:
In the TrackerObjectsMapper you prefix the var representing the value from the DB with `db`.

The `event` prefix isn't all that useful in the event persister. Its hard to understand what `eventDataValue` vs `dv` is.